README.md

```diff
@@ -28,6 +38,29 @@ await pushNotificationReceived(snapyrData);
 // after receiving a push interaction callback, track tapped metric to Snapyr
 const snapyrData = notification.data?.snapyr;
 await pushNotificationTapped(snapyrData);
+
+// --- In-App messaging section (optional) ---
+
+// listen to in-app messages from Snapyr 
+onSnapyrInAppMessage((inappMessage: SnapyrInAppMessage) => {
+    const actionToken = inappMessage.actionToken;
+    // example... stash this message in a useState so we can read its properties elsewhere
+    setCurrentInAppMessage(inappMessage);
+    if (inappMessage.actionType == "custom" && inappMessage.content.payloadType == "html") {
+        // example... use HTML template somewhere in your app
+        setHtmlContent(inappMessage.content.payload);
+    }
+});
+
+// track how users are interacting with your in-app message
+// ... after we've shown a message to the user...
+trackInAppMessageImpression(currentInAppMessage.actionToken);
+// ... after we've determined the user interacted with our message...
+trackInAppMessageClick(currentInAppMessage.actionToken, {exampleExtraProperty: "someId"});
+// ... if the user dismissed our message without interacting with it...
+trackInAppMessageDismiss(currentInAppMessage.actionToken);
```

Comment:
Does the app have to put in its own logic to only call this if no interactions happened prior? If so might be worth making that more explicit (either here or in the actual documentation).

Reply:
Yes - the tracking calls (will) have more details in the documentation but I'll try to clarify the comment here.